### PR TITLE
[vcpkg script] Find Python interpreter to used for scripts/buildsystems/osx/applocal.py

### PR DESCRIPTION
**Pull request by gladhorn**
**Describe the pull request**

- #### What does your PR fix?
  https://github.com/microsoft/vcpkg/issues/24696
  Running vcpkg on macOS would fail when there is no executable named "python" in path.
  The deployment script works with Python 2 and 3. On modern macOS machines Python 2 may not be available. Find either version and use it.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
   all (although this is specific to macOS, should not influence anything else), No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Unrelated

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by gladhorn** [comment]
I suppose the CI failure is unrelated? I cannot see the log in any case, any help appreciated :)

**Comment by autoantwort** [comment]
Fixes #24696

**Comment by dg0yt** [comment]
> I suppose the CI failure is unrelated? I cannot see the log in any case, any help appreciated :)

Unrelated, yes. (But the logs are freely accessible if needed.)

File: scripts/buildsystems/vcpkg.cmake [hunk 1]
```diff
@@ -603,8 +603,9 @@ function(add_executable)
                 )
             elseif(Z_VCPKG_TARGET_TRIPLET_PLAT MATCHES "osx")
                 if(NOT MACOSX_BUNDLE_IDX EQUAL -1)
+                    find_package (Python COMPONENTS Interpreter)
```
Comment (dg0yt) [inline scripts/buildsystems/vcpkg.cmake:606]: Please remove the extra space after `find_package`.

File: scripts/buildsystems/vcpkg.cmake [hunk 2]
```diff
@@ -603,8 +603,9 @@ function(add_executable)
...
                     add_custom_command(TARGET "${target_name}" POST_BUILD
-                        COMMAND python "${Z_VCPKG_TOOLCHAIN_DIR}/osx/applocal.py"
+                        COMMAND ${Python_EXECUTABLE} "${Z_VCPKG_TOOLCHAIN_DIR}/osx/applocal.py"
```
Comment (dg0yt) [inline scripts/buildsystems/vcpkg.cmake:608]: Please put quotes around `${Python_EXECUTABLE}`.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: scripts/buildsystems/vcpkg.cmake (hunk 2 above)
Comment (gladhorn) [reply scripts/buildsystems/vcpkg.cmake:608]: Makes sense :)

File: scripts/buildsystems/vcpkg.cmake (hunk 1 above)
Comment (gladhorn) [reply scripts/buildsystems/vcpkg.cmake:606]: Sure.

**Review by LilyWangLL — APPROVED** [review]

**Comment by dg0yt** [comment]
I think this PR could be merged despite the linux error. The linux error (in a special CI test port) seems to be an installation order problem with wxwidgets or gtk3. wxwidgets picked up EGL when it was built from source, but now EGL is not present when wxwidgets is installed from cache.
config:
~~~
-- Found wxWidgets: -L...;/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libEGL.a;...;-ldl (found version "3.2.1") 
~~~
build:
~~~
ninja: error: '/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libEGL.a', needed by 'exe', missing and no known rule to make it
~~~
So the problem won't go away without independent work, and it may affect other PRs.

**Pull request merged**